### PR TITLE
add dynamic-list-of-todos

### PR DESCRIPTION
**Pull request by DidenkoDenis306**
[DEMO LINK](https://DidenkoDenis306.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by And678 — CHANGES_REQUESTED** [review]
Good job, but when pressing enter on your input, the page needlessly reloads. Try using `event.preventDefault()` on form submit

File: src/components/TodoFilter/TodoFilter.tsx [hunk 1]
```diff
@@ -1,34 +1,74 @@
...
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {query && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```
Comment (dffUqp) [inline src/components/TodoFilter/TodoFilter.tsx:62]: let's fix this eslint issue:smiley:

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,100 +1,91 @@
...
+                    data-cy="selectButton"
+                    className="button"
+                    type="button"
+                    onClick={() => openTodo(todo)}
+                  >
+                    <span className="icon">
+                      <i className="far fa-eye" />
+                    </span>
+                  </button>
+                )}
```
Comment (dffUqp) [inline src/components/TodoList/TodoList.tsx:84]: I think here we change only icon, so let's replace this huge part of code with [classnames](https://www.npmjs.com/package/classnames) library

**Review by dffUqp — CHANGES_REQUESTED** [review]
LGTM but needs a few fixes:smiley:

**Review by 33Orange — APPROVED** [review]
Well done

**Pull request closed without merging**